### DISCUSSION TITLE
The scheduler does not appear to be running. Last heartbeat was received xxx minutes ago.

### DISCUSSION THREAD

**Discussion by ghost** [apache/airflow#21015] [Q&A]
I installed and reinstalled airflow several times now - OS = Ubuntu 16.04, Airflow version 2.2.3 -- It works fine, and opens, and importas the Dags, BUT after everything is set up, I keep getting the same message: "The scheduler does not appear to be running. Last heartbeat was received xx minutes ago. The DAGs list may not update, and new tasks will not be schedule". Then I kill all the processes and start over, and it works again, yet after some minutes again I get the same message. How can I make the scheduler not stop?

**Comment by potiuk** [comment]
* If you are using SequentialExecutor/SQLite - this is expected. Stop doing it. Switch to proper db/non-sequential executor. 

* If you are using "proper" DB and LocalExecutor - increase performance of your system (give it more memory probably) because likely it swaps out a lot to a disk which causes very slow general performance. As the last resort increase this: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#scheduler-health-check-threshold
* Finally - check if your scheduler is not "Really" killed by something (for example lack of memory)

**Reply by ghost** [reply]
On it.

**Reply by suniln1102** [reply]
Was your issue resolved? if yes, what changes did you make?

**Reply by mnsosa** [reply]
what is a "proper executor"? why it isn't proper anyway the default cfg?

**Reply by Taragolis** [reply]
> what is a "proper executor"?

Depends of requirements and resources. 

> why it isn't proper anyway the default cfg?

Nobody know which DB backed and executor user want to use. This decision is up to the user. Only SQLite and Sequential could be used without any configuration, so that is why it default

**Reply by potiuk** [reply]
> what is a "proper executor"?

Certainly "Sequential" executor is not. I mean "proper" as the executor that is marked as production ready.
